### PR TITLE
feat: add logging and bugsnag error reporting to sanity-graphql client

### PR DESCRIPTION
**Pull request by mcky**
## Description
Add bugsnag / console logging to failed sanity-graphql requests

❓ Will catching and re-throwing result in confusion from 2 bugsnag bugs? Instead of reporting should I be creating an OakError in the hopes it will propogate with metadata to the page-level bug?

❓ Should the same be added to the curriculum API?

## Issue(s)

Arose via ENG-308 (Make OWA  config resilient to Netlify setting empty strings on unspecified envs)

### CODE REVIEW

**Review by jimCresswell — APPROVED** [review]
Thanks @mcky ! This looks good to me, but I'd like @mantagen to take a look as well

File: src/node-lib/sanity-graphql/index.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import { GraphQLClient } from "graphql-request";
 
+import errorReporter from "../../common-lib/error-reporter";
 import getServerConfig from "../getServerConfig";
 
 import {
+  getSdk,
+  SdkFunctionWrapper,
```
Comment (mantagen) [inline src/node-lib/sanity-graphql/index.ts:8]: oh handy

**Review by mantagen — APPROVED** [review]
> ❓ Will catching and re-throwing result in confusion from 2 bugsnag bugs? Instead of reporting should I be creating an OakError in the hopes it will propogate with metadata to the page-level bug?

Still haven't formulated a consistent approach here, so what you've done is great

> ❓ Should the same be added to the curriculum API?

Good point, would make sense to add in one place per geenrated sdk

**Review by johnrobeds — APPROVED** [review]

**Pull request merged**